pysumma/ensemble.py

```diff
@@ -194,8 +198,10 @@ def monitor(self):
         Halt computation until submitted simulations are complete
         """
         simulations = self._client.gather(self.submissions)
-        for s in simulations:
-            self.simulations[s.run_suffix] = s
+        names = [n for n,s in self.simulations.items()]
+
+        for s,name in zip(simulations,names):
+            self.simulations[name] = s
```

Comment:
I'm guessing this was some sort of a bug before, but could you explain why this change was made?

Also, a complete nitpick, but you can do:
```
names = self.simulations.values()
```
instead of
```
names = [n for n,s in self.simulations.items()]
```

Reply:
Previously that the run_suffix was taken always as the  name. This was implemented to that the first key of the dictionary is used, allowing different ensemble naming. In my use case, allowing the same run_suffix to be used.

The simplified code that works is 
names = self.simulations.keys()

Reply:
Ah, that makes sense - thanks!